Arrays of dynamic size (documentation)

From http://cython.readthedocs.io/en/latest/src/userguide/language_basics.html it is unclear whether arrays of non-static length are allowed. Example:

```
cdef int eggs(unsigned long x):
    cdef int a[x]
```

Please make it explicit in the docs.

I don't think it depends on the C compiler. For example, if I do: 
```python 
cdef unsigned int a = 10
cdef int[a] b
```

We get the following output:
```
/mnt/c/Users/smith/Desktop/projects/testing_stuff $ runcython3 testing.pyx  -a

Error compiling Cython file:
------------------------------------------------------------
...

cdef unsigned int a = 10
cdef int[a] b
         ^
------------------------------------------------------------

testing.pyx:3:10: Not allowed in a constant expression
testing.c:1:2: error: #error Do not use this file, it is the result of a failed Cython compilation.
 #error Do not use this file, it is the result of a failed Cython compilation.
  ^
```

So the Cython compilation fails, independently of the C compiler.

I'll make a PR mentioning this somewhere.

--- a/docs/src/userguide/language_basics.rst
+++ b/docs/src/userguide/language_basics.rst
@@ -77,8 +77,9 @@ containers.
 
 Pointer types are constructed as in C, by appending a ``*`` to the base type
 they point to, e.g. ``int**`` for a pointer to a pointer to a C int.
-Arrays use the normal C array syntax, e.g. ``int[10]``.  Note that Cython uses
-array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
+Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known at
+at compile time for stack allocated arrays. Cython doesn't support variable lenght arrays from C99.
+Note that Cython uses array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
 whereas ``x[0]`` is.
 
 Also, the Python types ``list``, ``dict``, ``tuple``, etc. may be used for

--- a/docs/src/userguide/language_basics.rst
+++ b/docs/src/userguide/language_basics.rst
@@ -77,8 +77,8 @@ containers.
 
 Pointer types are constructed as in C, by appending a ``*`` to the base type
 they point to, e.g. ``int**`` for a pointer to a pointer to a C int.
-Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known at
-at compile time for stack allocated arrays. Cython doesn't support variable lenght arrays from C99.
+Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known
+at compile time for stack allocated arrays. Cython doesn't support variable length arrays from C99.
 Note that Cython uses array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
 whereas ``x[0]`` is.
 

